src/azul/__init__.py

```diff
@@ -991,23 +1012,49 @@ def _git_status(self) -> Mapping[str, str]:
         }
 
     @property
-    def lambda_git_status(self) -> Mapping[str, str]:
+    def lambda_git_status(self) -> dict[str, str]:
         return {
             'commit': self.environ['azul_git_commit'],
             'dirty': str_to_bool(self.environ['azul_git_dirty'])
         }
 
+    @property
+    def _aws_account_name(self) -> dict[str, str]:
+        return {
+            'azul_aws_account_name': self.aws_account_name
+        }
+
+    @property
+    def aws_account_name(self) -> str:
+        """
+        Retrieving the AWS account name from IAM violates the Lambda security
+        boundary. This methods avoids this by retrieving the account name from
+        the Lambda environment instead when invoked in a Lambda context.
```

Comment:
We could open up the policy, that's not the motivation for doing this. It's that we don't want to do the IAM round trip every time the function is executed.